Feat(spv 1493) update js client to a new version with split admin and user client

cain4chain opened the pull request:
- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

cain4chain committed:
feat(SPV-1493):update js-client to a new version with split admin and user client

chore(js-client): update js client version
chore(createClient): Define extended client types for admin and user API
chore(apis): Update all APIs to use new version for admin and user client from the new js-client library version 1.0.0-beta.32

cain4chain committed:
feat(SPV-1493): use tanstack store and react hooks for admin and user API

chore(clientStore): update client store using tanstack library for state management
chore(clientStore): use react hooks for user and admin API
chore(apis): update all API interactive files using the new client store version

chris-4chain commented on src/store/clientStore.ts:
Please check ESLint errors 👇

chris-4chain commented on src/store/clientStore.ts:
```diff
@@ -0,0 +1,77 @@
+};
+
+// Action to clear clients
+export const clearClients = () => {
+  clientStore.setState(() => ({
+    adminClient: null,
+    userClient: null,
+  }));
+};
+
+export const clearAdminClient = () => {
+  clientStore.setState((state) => ({
+    ...state,
+    adminClient: null,
+  }));
+};
+
+export const clearUserClient = () => {
+  clientStore.setState((state) => ({
+    ...state,
+    userClient: null,
+  }));
+};
+
+export const disconnectAdmin = () => clearAdminClient();
+export const disconnectUser = () => clearUserClient();
```
These functions seem to be unused 🙈

chris-4chain commented on src/store/clientStore.ts:
```diff
@@ -0,0 +1,77 @@
+import { Store, useStore } from '@tanstack/react-store';
+import { SpvWalletAdminClientExtended, SpvWalletUserClientExtended } from '@/contexts';
+
+interface ClientState {
+  adminClient: SpvWalletAdminClientExtended | null;
+  userClient: SpvWalletUserClientExtended | null;
+}
```
Should be rather `ClientStore`. 🤔

chris-4chain commented on src/components/AddAccessKeyDialog/AddAccessKeyDialog.tsx:
**General comment**
We should make a full transition and use the store and Admin or User API.

Like this:
![image](https://github.com/user-attachments/assets/7f55a3c5-5b11-4be1-986e-f53f9f11571f)

It would be great if we could get rid of the `useSpvWalletClient` and `SpvWalletContext` in favour of usage of the new Store. Because for now, it seems redundant.

cain4chain committed:
feat(SPV-1493) : add client store for admin and user seperation

feat(clientStore) : integrate client store into remaining endpoints
feat(lint) : update lint issues

cain4chain committed:
Merge branch 'main' into feat(SPV-1493)-update-js-client-to-a-new-version-with-split-admin-and-user-client

cain4chain committed:
feat(SPV-1493) : Update conflicts

chore(paymail_query) : update paymail query utility with new js client format

cain4chain committed:
Merge branch 'main' into feat(SPV-1493)-update-js-client-to-a-new-version-with-split-admin-and-user-client

cain4chain committed:
chore(update j client): js client version update

chore(update j client): js client version update to latest

cain4chain committed:
chore(install dep updates) : yarn lock updated

chore(install dep updates) : yarn lock updated to latest

chris-4chain commented on src/components/PaymailDeleteDialog/PaymailDeleteDialog.tsx:
**Manual tests result**

![image](https://github.com/user-attachments/assets/721e8b9d-0ec9-4d87-8d29-99cda782ad6f)

How to reproduce:
- Login as user
- Go to "paymails" tab
- Click "actions" in one of the rows

![image](https://github.com/user-attachments/assets/1a858901-0466-480b-9574-1ab019df5f2d)

cain4chain replied:
Thanks, I will update to remove user delete dialog abilities since this is only handled by Admin endpoints.

cain4chain committed:
feat(SPV-1493) :  Remove delete dialog for user access

chore(paymailsTabContent) : remove delete dialog access for user
fix(paymailDeleteDialog) : change paymail to id for delete paymail

cain4chain committed:
feat(SPV-1493) : Update styling

chore(prettier) : run prettier formatting

cain4chain committed:
feat(SPV-1493) - update admin status check

chore(clientStore) : add function for admin status check
